monai/losses/dice.py

```diff
@@ -298,5 +300,177 @@ def forward(self, input: torch.Tensor, target: torch.Tensor, smooth: float = 1e-
+        )
+        flat_target_extended = torch.unsqueeze(flat_target_extended, dim=1)
+
+        # Extract the vector of class distances for the ground-truth label at each voxel
+        m_extended = torch.gather(m_extended, dim=1, index=flat_target_extended)
+        m_extended = torch.squeeze(m_extended, dim=1)
+
+        # Compute the wasserstein distance map
+        wasserstein_map = m_extended * flat_proba
+
+        # Sum over the classes
+        wasserstein_map = torch.sum(wasserstein_map, dim=1)
+        return wasserstein_map
+
+    def compute_generalized_true_positive(
+        self, alpha: torch.Tensor, flat_target: torch.Tensor, wasserstein_distance_map
+    ):
+        """
+        Args:
+            alpha: generalised number of true positives of target class.
+            flat_target: the target tensor.
+            wasserstein_distance_map: the map obtained from the above function.
+        """
+        # Extend alpha to a map and select value at each voxel according to flat_target
+        alpha_extended = torch.unsqueeze(alpha, dim=2)
+        alpha_extended = alpha_extended.expand((flat_target.size(0), self.num_classes, flat_target.size(1)))
+        flat_target_extended = torch.unsqueeze(flat_target, dim=1)
+        alpha_extended = torch.gather(alpha_extended, index=flat_target_extended, dim=1)
+
+        # Compute the generalized true positive as in eq. 9
+        generalized_true_pos = torch.sum(alpha_extended * (1.0 - wasserstein_distance_map), dim=[1, 2],)
+        return generalized_true_pos
+
+    def compute_denominator(self, alpha: torch.Tensor, flat_target: torch.Tensor, wasserstein_distance_map):
+        """
+        Args:
+            alpha: generalised number of true positives of target class.
+            flat_target: the target tensor.
+            wasserstein_distance_map: the map obtained from the above function.
+        """
+        # Extend alpha to a map and select value at each voxel according to flat_target
+        alpha_extended = torch.unsqueeze(alpha, dim=2)
+        alpha_extended = alpha_extended.expand((flat_target.size(0), self.num_classes, flat_target.size(1)))
+        flat_target_extended = torch.unsqueeze(flat_target, dim=1)
+        alpha_extended = torch.gather(alpha_extended, index=flat_target_extended, dim=1)
+
+        # Compute the generalized true positive as in eq. 9
+        generalized_true_pos = torch.sum(alpha_extended * (2.0 - wasserstein_distance_map), dim=[1, 2],)
+        return generalized_true_pos
+
+    def compute_weights_generalized_true_positives(self, flat_target: torch.Tensor):
+        """
+        Args:
+            flat_target: the target tensor.
+        """
+        one_hot = F.one_hot(flat_target, num_classes=self.num_classes).permute(0, 2, 1).float()
+        volumes = torch.sum(one_hot, dim=2)
+        alpha = 1.0 / (volumes + 1.0)
+        return alpha
+
+
 dice = Dice = DiceLoss
 generalized_dice = GeneralizedDiceLoss
```

Comment:
We should add an alias here for `GeneralizedWassersteinDiceLoss`, so something like `wasserstein_dice = GeneralizedWassersteinDiceLoss`.

Reply:
yes adding the alias.